File: Arrays/Medium/189/189.py
from typing import List

class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n = len(nums)
        k %= n
        i, j = 0, n - k - 1
        while i <= j:
            nums[i], nums[j] = nums[j], nums[i]
            i += 1
            j -= 1
        i, j = n - k, n - 1
        while i <= j:
            nums[i], nums[j] = nums[j], nums[i]
            i += 1
            j -= 1
        nums.reverse()

File: Arrays/Medium/189/189.c
void rotate(int* nums, int numsSize, int k){
    k %= numsSize;
    int i = 0, j = numsSize - k - 1;
    while(i <= j){
        int temp = nums[i];
        nums[i] = nums[j];
        nums[j] = temp;
        i++;
        j--;
    }
    i = numsSize - k;
    j = numsSize - 1;
    while(i <= j){
        int temp = nums[i];
        nums[i] = nums[j];
        nums[j] = temp;
        i++;
        j--;
    }
    for(i = 0; i < numsSize / 2; i++){
        int temp = nums[i];
        nums[i] = nums[numsSize - i - 1];
        nums[numsSize - i - 1] = temp;
    }
}